crates/bloom-filter/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+use std::hash::{Hash, Hasher};
+use xxhash_rust::xxh3::Xxh3Builder;
+
+
+pub struct BloomFilter<const N: usize> {
+    bit_array: [u8; N],
+    num_hashes: usize,
+}
+
+
+impl<const N: usize> BloomFilter<N> {
+    pub fn new(num_hashes: usize) -> Self {
+        BloomFilter {
+            bit_array: [0; N],
+            num_hashes,
+        }
+    }
+
+    pub fn to_bit_array(self) -> [u8; N] {
+        self.bit_array
+    }
+
+    pub fn insert<T: Hash>(&mut self, item: &T) {
+        for i in 0..self.num_hashes {
+            let hash = self.hash(item, i as u64);
+            let byte_index = hash / 8;
+            let bit_index = hash % 8;
+            self.bit_array[byte_index] |= 1 << bit_index;
```

Comment:
Did you check that it corresponds to the endianness of the arrow lib? It would be more explicit to use [`set_bit_raw`](https://docs.rs/arrow-buffer/54.2.0/arrow_buffer/bit_util/fn.set_bit_raw.html) and `get_bit_raw` from `arrow_buffer`